R/SEIRDAge_interventions.R

```diff
@@ -0,0 +1,452 @@
+
+    object
+  })
+
+
+
+#' @describeIn SEIRDAge_interventions Method to simulate output using from model.
+#'
+#' Solves a system to ODEs which form an
+#' age-structured simple SEIR model. The system of equations for the time
+#' evolution of population fractions in Susceptible (S), Exposed (E), Infected
+#' (I), Recovered (R) and Dead (D) groups in a given age group indexed by i is
+#' given by
+
+#' where C is a contact matrix whose elements represents the
+#' contact between different age groups (rows) with age groups of
+#' people they come in contact with (columns). Inter(t) is the value at time t
+#' of the intervention protocol defined by
+#' the intervention parameters. This function relies on the package deSolve.
+#' The model encorperates the influence of non-pharmeceutical interventions (NPIs)
+#' which act to reduce the degree of social contact between individuals. This is is
+#' done by splitting any simulation of the model into 3 intervals and utilising a different
+#' contact matrix in each interval to simulate the introduction of NPIs through
+#' time. This function relies on the
+#' package deSolve to numerically integrate the set of equations above.
+#'
+#' @param object An object of the class SEIRDAge_interventions.
+#' @param times (vector) time sequence over which to solve the model.
+#'        Must be of the form seq(t_start,t_end,by=t_step). Default time series
+#'        is seq(0, 100, by = 1).
+#'
+#' @param solve_method A string indicating the chosen numerical integration
+#' method for solving the ode system. Default is `lsoda` which is also the
+#' default for the ode function in the deSolve package used in this function.
+#'
+#' @return a data frame containing the time vector and time series of S, R, I and
+#' D population fractions for each age group outputs with incidence numbers
+#' for each age group.
+#'
+#' @export
+#' @importFrom magrittr "%>%"
+#' @importFrom rlang .data
+setMethod(
+  "run", 'SEIRDAge_interventions',
+  function(object, times, solve_method = 'lsoda') {
+
+    # error if times is not a vector or list of doubles
+    if(!is.double(times)){
+      stop('Evaluation times of the model storage format must be a vector.')
+    }
+    if (is.null(unlist(object@transmission_parameters)))
+      stop("Transmission parameters must be set before running.")
+    if (is.null(unlist(object@initial_conditions)))
+      stop("Initial conditions must be set before running.")
+    if (is.null(unlist(object@intervention_parameters)))
+      stop("Intervention parameters must be set before running.")
+
+    # fetch number of age catagories
+    age <- object@n_age_categories
+
+    # set initial state vector
+    state <- c(S = object@initial_conditions$S0,
+               E = object@initial_conditions$E0,
+               I = object@initial_conditions$I0,
+               R = object@initial_conditions$R0,
+               D = object@initial_conditions$D0,
+               cc = rep(0, age))
```

Comment:
Does "cc" mean those individuals who are cocooning or something else? In any case, we probably want to use capital letters like for the other states?

Reply:
"cc" indicates the cumulative number of cases over time. It is "C" in the basic model, but in models with contact matrices, we name the contact matrices as "C" in ODEs and "cc" for cumulative cases instead. I simply took the same namings from other age-structured models.

Do we want the names be changed?